src/components/TextInput/BaseTextInput.js

```diff
@@ -18,387 +18,387 @@ import getSecureEntryKeyboardType from '../../libs/getSecureEntryKeyboardType';
-        const textInputContainerStyles = _.reduce(
-            [
-                styles.textInputContainer,
-                ...this.props.textInputContainerStyles,
-                this.props.autoGrow && StyleUtils.getWidthStyle(this.state.textInputWidth),
-                !this.props.hideFocusedState && this.state.isFocused && styles.borderColorFocus,
-                (this.props.hasError || this.props.errorText) && styles.borderColorDanger,
-                this.props.autoGrowHeight && {scrollPaddingTop: 2 * maxHeight},
-            ],
-            (finalStyles, s) => ({...finalStyles, ...s}),
-            {},
-        );
-        const isMultiline = this.props.multiline || this.props.autoGrowHeight;
-
-        return (
-            <>
-                <View>
-                    <PressableWithoutFeedback
-                        onPress={this.onPress}
-                        focusable={false}
-                        accessibilityLabel={this.props.label}
+    };
+
+    const togglePasswordVisibility = useCallback(() => {
+        setPasswordHidden((prevState) => !prevState.passwordHidden);
+    }, []);
+
+    const storePrefixLayoutDimensions = useCallback((event) => {
+        setPrefixWidth(Math.abs(event.nativeEvent.layout.width));
+    }, []);
+
+    // eslint-disable-next-line react/forbid-foreign-prop-types
+    const inputProps = _.omit(props, _.keys(baseTextInputPropTypes.propTypes));
+    const hasLabel = Boolean(props.label.length);
+    const isEditable = _.isUndefined(props.editable) ? !props.disabled : props.editable;
+    const inputHelpText = props.errorText || props.hint;
+    const placeholder = props.prefixCharacter || isFocused || !hasLabel || (hasLabel && props.forceActiveLabel) ? props.placeholder : null;
+    const maxHeight = StyleSheet.flatten(props.containerStyles).maxHeight;
+    const textInputContainerStyles = StyleSheet.flatten([
+        styles.textInputContainer,
+        ...props.textInputContainerStyles,
+        props.autoGrow && StyleUtils.getWidthStyle(textInputWidth),
+        !props.hideFocusedState && isFocused && styles.borderColorFocus,
+        (props.hasError || props.errorText) && styles.borderColorDanger,
+        props.autoGrowHeight && {scrollPaddingTop: 2 * maxHeight},
+    ]);
+    const isMultiline = props.multiline || props.autoGrowHeight;
+
+    return (
+        <>
+            <View>
+                <PressableWithoutFeedback
+                    onPress={onPress}
+                    focusable={false}
+                    accessibilityLabel={props.label}
+                    style={[props.autoGrowHeight && styles.autoGrowHeightInputContainer(textInputHeight, maxHeight), !isMultiline && styles.componentHeightLarge, ...props.containerStyles]}
+                >
+                    <View
+                        // When autoGrowHeight is true we calculate the width for the textInput, so It will break lines properly
```

Comment:
NAB
```suggestion
                        // When autoGrowHeight is true we calculate the width for the textInput, so it will break lines properly
```